RTClib.cpp

```diff
@@ -746,6 +746,168 @@ TimeSpan::TimeSpan(int16_t days, int8_t hours, int8_t minutes, int8_t seconds)
 /**************************************************************************/
 TimeSpan::TimeSpan(const TimeSpan &copy) : _seconds(copy._seconds) {}
 
+/*!
+    @brief  Create a new TimeSpan from an iso8601 formatted period string
+
+    If the string is entirely malformed (doesn't begin with P or -) this will
+    construct a TimeSpan of 0 seconds. If parsing fails before the end of the
+    string, this constuctor will interpret its argument as the longest
+    well-formed prefix string.
+
+    For example, the string P5M10~ will parse the same as P5M
+
+    @param iso8601 the formatted string
+*/
+TimeSpan::TimeSpan(char *iso8601) : _seconds(0) {
+  int32_t sign = 1;
+  char *cursor = iso8601;
+  if (*cursor == '-') {
+    sign = -1;
+    cursor++;
+  }
+  if (*cursor == 'P') {
+    cursor++;
+  } else {
+    return;
+  }
+
+  char *rest;
+  long num = strtol(cursor, &rest, 10);
```

Comment:
If there are no digits, `strtol()` returns 0, and this case is not detected: this constructor parses "PDTHM30S" as 30&nbsp;seconds. Not sure that string is valid, the Wikipedia page is not very clear on this. [The Java doc on `Duration.parse()`](https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html#parse-java.lang.CharSequence-) is more specific: it states (emphasis mine) “The number part of each section must consist of **one or more** ASCII digits.”

Reply:
I'm fine with this based on the [robustness principle](https://en.wikipedia.org/wiki/Robustness_principle). Also I found [this pdf](https://dotat.at/tmp/ISO_8601-2004_E.pdf) online. Not sure how legit it is.